[Feat/#321] 점검시간 다이얼로그 국제화 및 포맷 수정을 진행합니다.

SYAAINN opened the pull request:
## 📌 ISSUE
<!--이슈 번호 및 제목을 적어주세요!-->
closed #321 

## 📄 Work Description
<!--어떤 작업을 했는지 작성해주세요!-->
- 점검시간 다이얼로그 string 추출로 국제화
- 분 단위도 나타낼 수 있도록 언어별 포맷 수정
- 점검시간을 유저 타임존에 맞추어 변환

## ✨ PR Point
<!--리뷰어가 특별히 봐주었으면 하는 부분이 있다면 작성해주세요!! 코드를 넣어도 됩니다!-->
원래는 remoteConfig에서 가져온 2025-08-11T18:00:00 형식의 점검시간을 가공 후 뷰모델로 넘겨주었는데
언어 분기 처리를 위한 LanguageProvider 때문에 layer간 의존성이 역전되어,, 방식을 조금 변경했습니다.
1. data layer (AppUpdateChecker) 에서는 raw한 형태로 fetch 하여 전달하는 역할만 합니다.
2. 뷰모델에서 시작시간/종료시간을 받은 후 LanguageProvider에 넘깁니다
3. LanguageProvider에서 언어별, 타임존별로 알맞은 포맷으로 가공합니다.

## 📸 ScreenShot/Video
<!--스크린샷이나 영상을 첨부해주세요! 생략 하셔도 무방합니다!-->
<img width="394" height="384" alt="스크린샷 2025-08-14 22 41 17" src="https://github.com/user-attachments/assets/5904f76e-465e-4e4f-8964-5d584bc598ce" />


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->

## Summary by CodeRabbit

* New Features
  * Added a localized maintenance dialog with dynamic time formatting (supports Korean and English).
  * Splash screen now displays localized maintenance window when applicable.
* Refactor
  * Improved Home screen data loading for more reliable calendar and diary updates; smoother deletion flow.
* Changes
  * Notification permissions: Home screen requests Android 13+ notification permission when needed; Time Reminder no longer prompts for notification permission.
* Chores
  * App version updated to 1.5.0 (build 29).

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

SYAAINN committed:
[REFACTOR/#321] 점검 다이얼로그 string 추출을 통한 국제화를 진행합니다.

SYAAINN committed:
[REFACTOR/#321] 점검 시작시간/종료시간을 "2025-08-11T18:00:00" 형태 그대로 반환합니다.

SYAAINN committed:
[REFACTOR/#321] 언어에 맞는 점검시간 포맷을 제공하는 함수를 구현합니다.

SYAAINN committed:
[REFACTOR/#321] 변경사항을 적용합니다.

SYAAINN committed:
[REFACTOR/#321] 점검시간을 유저의 타임존에 맞춰 변환한 후 보여줄 수 있도록 합니다.

SYAAINN committed:
[CHORE/#321] 영어버전도 24시간 형식으로 통일합니다.

MoonsuKang commented on app/src/main/java/com/sopt/clody/data/remote/appupdate/AppUpdateCheckerImpl.kt:
```diff
@@ -37,26 +36,18 @@ class AppUpdateCheckerImpl @Inject constructor(
-        val startText = formatDateTimeWithDayOfWeek(start)
-        val endText = formatDateTimeWithDayOfWeek(end)
-        return "$startText ~ $endText"
+        return nowServer.isAfter(startZ) && nowServer.isBefore(endZ)
```
이렇게도 될 것 같아요
```suggestion
        return nowServer in startZ..endZ
```

SYAAINN replied:
기존 방식은 초과/미만이고 제안해주신 방식은 이상/이하라고 하네요! 이상/이하가 더 자연스러울 것 같아 반영했습니다!

MoonsuKang commented on app/src/main/java/com/sopt/clody/data/remote/appupdate/AppUpdateCheckerImpl.kt:
```diff
@@ -37,26 +36,18 @@ class AppUpdateCheckerImpl @Inject constructor(
-        val now = LocalDateTime.now()
-        return now.isAfter(start) && now.isBefore(end)
-    }
+        val serverZone = ZoneId.of("Asia/Seoul")
```
companion object로 분리하는게 더 좋아보입니다~

MoonsuKang commented on app/src/main/java/com/sopt/clody/presentation/utils/language/LanguageProviderImpl.kt:
```diff
@@ -13,6 +15,25 @@ class LanguageProviderImpl @Inject constructor() : LanguageProvider {
+            val startLdt = LocalDateTime.parse(start, DateTimeFormatter.ISO_LOCAL_DATE_TIME)
+            val endLdt = LocalDateTime.parse(end, DateTimeFormatter.ISO_LOCAL_DATE_TIME)
+
+            if (isKorean()) {
```
함수명이 `getInspectionTimeText` 이라면 formatting 하는 부분은 별도의 함수로 분리하는게 좋을 것 같아유

SYAAINN replied:
단일 책임 원칙 고려하여 포맷핑 부분 분리했습니다.

MoonsuKang commented on app/src/main/java/com/sopt/clody/presentation/utils/language/LanguageProviderImpl.kt:
```diff
@@ -13,6 +15,25 @@ class LanguageProviderImpl @Inject constructor() : LanguageProvider {
+            val endLdt = LocalDateTime.parse(end, DateTimeFormatter.ISO_LOCAL_DATE_TIME)
+
+            if (isKorean()) {
+                val koPattern = DateTimeFormatter.ofPattern("M/d(E) HH시mm분", Locale.KOREAN)
```
요런 패턴들도 companion object로 분리하는게 좋습니다.
실무에선 다 분리하는 편 이에유

MoonsuKang reviewed (approved):
수고하셨슴다~ 리뷰는 그냥 읽어만 주셔도 되유

SYAAINN committed:
[CHORE/#321] 리뷰 사항을 반영합니다.
- isUnderInspection 함수에서 점검시간 검사를 초과/미만에서 이상/이하로 개선 + companion object 분리
- getInspectionTimeText 함수에서 단일 책임 원칙에 따라 포맷팅 함수 분리 + companion object 분리
